internal/provider/resource_access_jaas_role_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the Apache License, Version 2.0, see LICENCE file for details.
+
+package provider
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	jimmnames "github.com/canonical/jimm-go-sdk/v3/names"
+	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/juju/names/v5"
+
+	internaltesting "github.com/juju/terraform-provider-juju/internal/testing"
+)
+
+// This file has bare minimum tests for role access
+// verifying that users, service accounts and roles
+// can access a role. More extensive tests for
+// generic jaas access are available in
+// resource_access_jaas_model_test.go
+
+func TestAcc_ResourceJaasAccessRole(t *testing.T) {
+	OnlyTestAgainstJAAS(t)
+	// Resource names, note that role two has access to role one.
+	RoleAccessResourceName := "juju_jaas_access_role.test"
+
+	roleOneResourceName := "juju_jaas_role.test"
+	accessSuccess := "assignee"
+	accessFail := "bogus"
+	user := "foo@domain.com"
+	roleOneName := acctest.RandomWithPrefix("role1")
+	svcAcc := "test"
+	svcAccWithDomain := svcAcc + "@serviceaccount"
+
+	// Objects for checking access
+	RoleRelationF := func(s string) string { return jimmnames.NewRoleTag(s).String() }
+	roleOneCheck := newCheckAttribute(roleOneResourceName, "uuid", RoleRelationF)
+	UserTag := names.NewUserTag(user).String()
+	svcAccTag := names.NewUserTag(svcAccWithDomain).String()
+
+	// Test 0: Test an invalid access string.
+	// Test 1: Test adding a valid set user, role and service account.
+	// Test 2: Test importing works.
```

Comment:
nit: s/Test 0/Step 0/ (etc) - Test implies tear down between, this test is a set of sequential steps.